test/api/OperatorApi/orderStatus.test.ts

```diff
@@ -0,0 +1,248 @@
+import { RawOrder } from 'api/operator'
+import { getOrderStatus } from 'api/operator/utils'
+
+import { ORDER } from '../../data'
+import { mockTimes, DATE } from '../../testHelpers'
+
+function _getCurrentTimestamp(): number {
+  return Math.floor(Date.now() / 1000)
+}
+
+function _getPastTimestamp(): number {
+  return Math.floor(DATE.getTime() / 1000) - 1
+}
+
+// mockTimes set's Date.now() to DATE const in the test context
+beforeEach(mockTimes)
+
+describe('Filled status', () => {
+  describe('Buy order', () => {
+    test('Filled, within epsilon', () => {
+      const order: RawOrder = { ...ORDER, kind: 'buy', buyAmount: '10000', executedBuyAmount: '9999' }
+
+      expect(getOrderStatus(order)).toEqual('filled')
+    })
+    test('Filled, exact amount', () => {
+      const order: RawOrder = { ...ORDER, kind: 'buy', buyAmount: '100', executedBuyAmount: '100' }
+
+      expect(getOrderStatus(order)).toEqual('filled')
+    })
+    test('Filled, surplus', () => {
+      const order: RawOrder = { ...ORDER, kind: 'buy', buyAmount: '1000', executedBuyAmount: '1001' }
+
+      expect(getOrderStatus(order)).toEqual('filled')
+    })
+    test('Filled, not yet expired', () => {
+      const order: RawOrder = {
+        ...ORDER,
+        kind: 'buy',
+        buyAmount: '100',
+        executedBuyAmount: '100',
+        validTo: _getCurrentTimestamp(),
+      }
+
+      expect(getOrderStatus(order)).toEqual('filled')
+    })
+    test('Filled, sell amount does not affect output', () => {
+      const order: RawOrder = {
+        ...ORDER,
+        kind: 'buy',
+        buyAmount: '100',
+        executedBuyAmount: '100',
+        sellAmount: '100',
+        executedSellAmount: '1100',
+      }
+
+      expect(getOrderStatus(order)).toEqual('filled')
+    })
+  })
+  describe('Sell order', () => {
+    test('Filled, within epsilon', () => {
+      const order: RawOrder = { ...ORDER, kind: 'sell', sellAmount: '10000', executedSellAmount: '9999' }
+
+      expect(getOrderStatus(order)).toEqual('filled')
+    })
+    test('Filled, exact amount', () => {
+      const order: RawOrder = { ...ORDER, kind: 'sell', sellAmount: '100', executedSellAmount: '100' }
+
+      expect(getOrderStatus(order)).toEqual('filled')
+    })
+    test('Filled, surplus', () => {
+      const order: RawOrder = { ...ORDER, kind: 'sell', sellAmount: '1000', executedSellAmount: '1001' }
```

Comment:
This is only possible if `feeAmount >= 1`

Reply:
Agree, although I would leave the test, but maybe rename it to `Filled, surplus (not really possible in contracts)`. I mean, the alternative would be to be to throw in these cases, and it feel unnecessarily strict. It's nice to say in these case the order is filled.

Reply:
If we insist, I wouldn't be opposed to throw, but I think is just too much for the states to add these validations. I think is enough to be resilient to these.

Reply:
To be clear, I wasn't suggesting that the code was handling this incorrectly, it was more about the fact that the test isn't really testing a "filled with surplus" state, what would be something like:
```ts
{
  ...ORDER,
  kind: 'sell',
  sellAmount: '1000',
  executedSellAmount: '1000',
  buyAmount: '1000',
  executedBuyAmount: '1337',
}
```

So maybe the test should be named something along the lines of `accepts invalid over-executed trades`?

Reply:
Since the surplus does not affect at all the order status, I'm in favor or removing surplus tests (besides the fact that they represent an invalid state on the backed)